Sources/SwiftCentrifuge/Client.swift

```diff
@@ -233,6 +233,34 @@ public class CentrifugeClient {
         self.subscriptions.append(sub)
         return sub
     }
+
+    /**
+     Try to get Subscription from internal client registry. Can return nil if Subscription
+     does not exist yet.
+     - parameter channel: String
+     - returns: CentrifugeSubscription?
+     */
+    public func getSubscription(channel: String) -> CentrifugeSubscription? {
+        defer { subscriptionsLock.unlock() }
+        subscriptionsLock.lock()
+        return self.subscriptions.first(where: { $0.channel == channel })
+    }
+
+    /**
+     * Say Client that Subscription should be removed from internal registry. Subscription will be
+     * automatically unsubscribed before removing.
+     - parameter channel: String
+     */
+    public func removeSubscription(channel: String) {
```

Comment:
Hello, thanks for pr! Is it possible to make this function match the same from Java client analogue:
```java
   public void removeSubscription(Subscription sub) {
        synchronized (this.subs) {
            sub.unsubscribe();
            if (this.subs.get(sub.getChannel()) != null) {
                this.subs.remove(sub.getChannel());
            }
        }
    }
```

I mean accept Subscription object instead of channel, and also removing should unsubscribe from channel.

I also think that removing should make Subscription object unusable: move Subscription to CLOSED state so that calling `sub.Subscribe()` method resulted in error for closed Subscription. This is optional for now – but worth adding in future versions I think.

Reply:
Changed the API as you suggested.

It's a great idea to make a subscription unusable after removing it. I think it would be better to do it in a separate PR. A lot more code changes and it'll require thorough testing.

This and storing subscriptions in a map (like in the android lib) would be nice to have :)

Reply:
There is actually a similar [ongoing pull request](https://github.com/centrifugal/centrifuge-go/pull/42) in `centrifuge-go` repo – so maybe eventually it makes sense to change API adding `subscription.Close()` method (here and Dart/Java clients) replacing `removeSubscription`. But for now, let's go with `client.removeSubscription`.

Regarding keeping subs in a map – feel free to send separate pr.

I will try to evaluate changes here this weekend, seems nice at quick glance. Thanks!